### ISSUE TITLE
Ability to hide reposts and quote posts per account

### ISSUE THREAD

**Issue by mschwendener** [bluesky-social/social-app#1116]
**Is your feature request related to a problem? Please describe.**

I know that I can hide reposts and quote posts via my settings. But I'd like to be able to decide that for each account I'm following.

**Describe the solution you'd like**

A setting to hide reposts yes/no for each account I'm following. 
 
**Describe alternatives you've considered**

It would be nice to also have the setting per account for quote posts, but that's less important for me.
 
**Additional context**

I used that a lot on twitter: the decision depends on how much the person reposts, and if I find their reposts interesting or not.

**Comment by mschwendener** [comment]
Looking forward to this :-)

**Comment by mschwendener** [comment]
In the meantime I'm using https://bsky.app/profile/skyfeed.xyz/feed/only-posts, which hides reposts.

**Comment by dariaambrose** [comment]
as a new bluesky user, I love my friends but some of them repost too much or repost bad posts. I would like to mute their reposts please.

**Comment by gaearon** [comment]
We're definitely aware of the importance of this feature — blocked on a few things on the backend so we can't prioritize it right now, but we're planning to get to it as soon as those blockers are solved.

**Comment by udany** [comment]
Congrats on the fast pace bsky's been evolving, just coming here to reinforce the request for this!

Hope thos blockers get solved soon ♥

**Comment by Go1den** [comment]
This would be a great addition. OnlyPosts is helpful in reducing the clutter of reposts, but I'd like more control over my feed than all-or-nothing.

**Comment by NanoBunTV** [comment]
Looking forward to this! 😊

**Comment by v317** [comment]
Looking forward to this feature!

**Comment by QueenParity** [comment]
Looking forward to this!

**Comment by HeroProc** [comment]
I will just say this specific feature is pretty vital for a lot of folks trying to migrate over. I have folks I follow that I want to see on my feed but I dont need to see the dozen reposts they propagate each day (for various reasons). I'm finding myself holding off following a lot of folks until this is implemented.

**Comment by Kazard** [comment]
Looking forward to this feature

**Comment by Crispy-o** [comment]
Totally understand there being more pressing concerns, but this would be a VERY nice feature on top of all the curation tools the dev team has given us. Eager to see this be implemented!

**Comment by ConnorKrammer** [comment]
Related, and immediately actionable given existing features: I could not find a feed that displays one's own posts with reposts filtered out. It would be nice if I could add this to my feeds list so that people who want to check out my profile can take a look to see what I post, without seeing what I repost. This is currently possible if I make a custom feed of my own, but it doesn't make too much sense for everyone to do this independently.

This has good synergy with an eventual "filter out reposts" option, because this feed would basically represent what they'd be seeing if they do so.

**Comment by Jeanfeels** [comment]
+1 for this. I know you guys are working hard behind the scenes, i just hope that whatever backend issue is preventing this from being implemented is given attention soon as fully half of my feed is tainted by bad reposts

**Comment by xnuk** [comment]
Could we just add '👍' emoji reaction at the top of this issue, instead of just commenting "+1" or "looking forward" or whatever? I've subscribed this issue for a while ago, and some recent comments feel like redundant.

**Comment by teatwig** [comment]
I'd like even more fine grained control if possible so we could implement a generic solution that defines the "type" of posts to hide. 

- hide reposts: For accounts that repost a lot of things I'm not interested in, but I'm still interested in their actual posts. 
- hide quote posts: See above. 
- hide text posts: This hides all posts of an account that don't have any media attached to them. For example I follow an artist that also posts a lot of personal things. This would allow me to only see their art. Most importantly this would not apply to posts that mention yourself. 
- hide media posts: Opposite of "hide posts". Probably less useful but good to cover all cases. 
- hide all posts: Pretty much identical to the current "mute" with the exception that it does not include posts that mention yourself. My case would be that someone I follow reposts a specific account a lot, but I'm still interested in their other reposts. This would allow me to to reduce the amount of posts I see while still being able to interact with that account. 

I think my main point is that the current implementation of "mute" heavily overlaps with blocking, at least for the person who initiated the mute. 
Both lead to no longer seeing any posts of the muted/blocked account. 

Adding comprehensive filter options could easily solve this (not saying that it would be easy to implement). 

The main issue I see is that these might be too many options to display in the app, so it could make sense to put them all behind a "Hide …" option.

**Comment by Hutchyy9** [comment]
Any update on this being added?

**Comment by projetotolkien** [comment]
I really wish this feature existed. Sure, you can turn off reposts for everyone, but it would be great to have the option to disable them for specific users who tend to overdo it. As someone mentioned before, I love my friends, but some of them repost way too much. I’d prefer to turn off only their reposts rather than disabling reposts for everyone (which is why the "only posts" feed doesn’t fully solve the issue). This was a feature I frequently used back on Twitter.

**Comment by projetotolkien** [comment]
I really wish this feature existed (it seems to have been on the roadmap for over a year now). Sure, you can turn off reposts for everyone, but it would be great to have the option to disable them for specific users who tend to overdo it. As someone mentioned before, I love my friends, but some of them repost way too much. I’d prefer to turn off only their reposts rather than disabling reposts for everyone (which is why the "only posts" feed doesn’t fully solve the issue). This was a feature I frequently used back on Twitter.

**Comment by kaolincash** [comment]
i am *desperate* for this feature.

**Comment by uncenter** [comment]
> I know I'm going to get downvoted, but seriously, this is probably Bluesky's biggest feature gap right now and it's killing the experience, especially as a flood of new people repost *everything*. Any updates on priority, timeline -- anything at all -- would be greatly appreciated. 
> 
> 
> 
> Right now, my choices are either turn off reposts altogether (damaging my feed and experience) or just give up and leave.

Fwiw you can probably write a quick user script to hide reposts for some users on the web version for now, though I get your frustration.

**Comment by liamness** [comment]
As much as I don't think comments saying how important the feature is are useful (clearly the team are already well aware, they just also have lots of other very important tasks on their plate, and there are other ways to express this without emailing everyone subscribed to this issue), I do want to do something more constructive than just give them a thumbs down. As @uncenter says, it should be possible to filter reposts out if using the web version at least. Here's a filter rule (compatible with uBlock Origin, but probably other content filtering tools) which will work to hide reposts from a given account:

`bsky.app##div[data-testid^="feedItem-by-"]:has-text(/Reposted by you/)`

This blocks your own reposts. Obviously that probably isn't what you want, so change the text in the rule from "you" to the display name of the account in question. This is obviously less than ideal, being a manual process, and needing a line for each account you feel is too repost-happy. It would also break if an account changes their display name (or if you followed two accounts with the same display name, e.g. let's say if they both just use their first name), or obviously if the markup changes. But hopefully this is at least a way to make the site usable for some people, while they wait for a real solution?

**Comment by DavidWBrooks** [comment]
> bsky.app##div[data-testid^="feedItem-by-"]:has-text(/Reposted by you/)`

For those of us somewhat technically aware: great! For those of us not 100% up to speed on BlueSky* configuration, where do I put this?
(*) I wish I could abbreviate BlueSky the same way I abbreviate WikiPedia.

**Comment by gaearon** [comment]
>I know I'm going to get downvoted, but seriously, this is probably Bluesky's biggest feature gap right now and it's killing the experience, especially as a flood of new people repost everything. Any updates on priority, timeline -- anything at all -- would be greatly appreciated.

Please believe me, everyone on the team is aware this is a huge missing feature. The backend team is at capacity at the moment and this feature in particular is gnarly to implement within the constraints of how our backend works today. In other words, this is very much a known gap but no amount of asking here will help bump it up. I want it yesterday too.

**Comment by djb-rh** [comment]
Additional request as part of this: there are accounts that I don't want to block all reposts, I just want to block when they repost THEMSELVES. So many in the media, in particular "repost for the morning crowd" something they already posted. I keep up with my entire timeline generally so I NEVER want to see a post twice.

**Comment by Stealrull** [comment]
This is the one thing keeping me from fully swapping to bluesky, my feed is completely reposts it sucks, hope this comes sooner then later but as it's been a request since 2023 I don't hold much hope...

**Comment by vic-t** [comment]
> Additional request as part of this: there are accounts that I don't want to block all reposts, I just want to block when they repost THEMSELVES. So many in the media, in particular "repost for the morning crowd" something they already posted. I keep up with my entire timeline generally so I NEVER want to see a post twice.

This. There should be a distinction between own reposts and third-party reposts. For me personally, the latter is usually fine but if I can block both of them for particular users, that would be ideal. But I never need to see own reposts, they're just a nuisance.

**Comment by joerdav** [comment]
> Additional request as part of this: there are accounts that I don't want to block all reposts, I just want to block when they repost THEMSELVES. So many in the media, in particular "repost for the morning crowd" something they already posted. I keep up with my entire timeline generally so I NEVER want to see a post twice. 

This is also why I stumbled upon this issue. Not interested in seeing people bump their own posts.

**Comment by dean-e-peterson** [comment]
When this feature is implemented, please consider adding a FAQ entry describing how to use it.  It seems appropriate since this feature is in high demand.

**Comment by SexyTomAfterDark** [comment]
Please keep us updated on this. We really need this feature. 
I don't wanna block ALL reposts. Just from specific users.

**Comment by Joabyjojo** [comment]
Excited to be approaching the 2 year anniversary of this request! Would also really like to only block when people repost themselves, but will settle for the ability to block all reposts from specific users.

**Comment by RangePup** [comment]
To everyone asking for a feed without Reposts rather than just turning off one persons, what this feature request actually is about. That already exists. https://bsky.app/profile/did:plc:tenurhgjptubkk5zf5qhi3og/feed/only-posts

**Comment by SexyTomAfterDark** [comment]
@RangePup 
We wan't to disable the reposts of SPECIFIC users. That is what this request is about.
We know we can disable ALL reposts via feed.

**Comment by RaphaelArctic** [comment]
I can see the tag "on the roadmap", but is there anything more detailed to follow? I could really use this feature and have also been waiting on it since i first got invited to the private bluesky period.

**Comment by Kirkman** [comment]
Have really, really wanted this since the beginning.

**Comment by Mementh** [comment]
100000% i need my posts to not be there, but everything else..  seriously

**Comment by Go1den** [comment]
It's almost 2026 and this still isn't implemented?

**Comment by TheCrossBoy** [comment]
it would be *really* nice to have this feature! but it would also be extremely helpful to see more context than just "on the roadmap" and "there are blockers". where on the roadmap is it? what specific issues are blocking this?

being able to follow along with those as well would be extremely helpful here to track the progress on this.

**Comment by klazoklazo** [comment]
This comment: https://github.com/bluesky-social/social-app/issues/1116#issuecomment-2523885825 mentions a workaround possible using uBlock Origin filters, though I wanna note there's a little room for improvement AFAIK:

`bsky.app##:matches-path(/)div[data-testid^="feedItem-by-"]:has(a[href="/profile/PLACEHOLDER.bsky.social"]):has(div:has-text("/^Reposted by/"))`

Instead of relying on a display name which is a lot more volatile on microblogging platforms, this tries to read the user's handle, which generally tends to be more stable. This then does a couple of checks, including excluding the filter from someone's userpage. If you want to remove reposts from someone entirely, including their own page, you can replace the regex from the beginning with the original, more general hostname like this:

`bsky.app##div[data-testid^="feedItem-by-"]:has(a[href="/profile/PLACEHOLDER.bsky.social"]):has(div:has-text("/^Reposted by/"))`

It's a similar procedure to liamness' comment, just paste the cosmetic filter into your "My filters" tab inside uBlock Origin, and replace the all caps PLACEHOLDER.bsky.social with your handle of choice. If who you're muting has a different domain from "bsky.social", Remember to replace the whole handle, including the "bsky.social" part.

**Comment by SneakyAzWhat** [comment]
on the roadmap for 2026? a man can dream

**Comment by Krovasan** [comment]
would love this feature, any update on it?

**Comment by mateosantosdev** [comment]
Any update? It was added to the roadmap in 2023 :(

**Comment by Mementh** [comment]
> Any update? It was added to the roadmap in 2023 :(

been using on fire fox https://addons.mozilla.org/en-US/firefox/addon/bluesky-repost-hider/
put in your name there or the account you don't want to see reposts from. its blocked.. it has to edit on the fly

**Comment by aknightzero** [comment]
Please for the love of god, bsky mods, implement a feature to Turn Off Reposts For User. Having to wade through hundreds of reposts is making me actively close the app and push me to stay away for good.

**Comment by RipPurr2** [comment]
Adding my vote to this feature request. In my case it helps get fun comedy and interest areas, and would effectively turn off tragic reposts from countries I don't live in. Sounds a bit harsh, but I use Bluesky for joy and this feature would help maintain that.

**Comment by RubeHicksCube** [comment]
Adding my vote for this. I like to find new users and just see what they're about. Not what everyone else is up to that they reposted from. Buried somewhere deep is a single post from the user but millions of reposts. I won't follow that user back if that's all I see.

**Comment by DIPALOT** [comment]
Same, i reaally need this feature

**Comment by isopodshuffle** [comment]
"On the roadmap" when it's actually so far off you need star charts to find it

**Comment by SpriterGors** [comment]
I'm also waiting for it, I even mentioned it in https://github.com/bluesky-social/social-app/issues/6922 and that was in 2024!!

**Comment by SnoillerFitzroy** [comment]
It's been almost 3 years. This is an UNACCEPTABLY long wait for something that should have been in since day 1.

And where's the ability to remove oneself from lists? Plenty of users have pointed out that moderation lists can themselves be used as mass-harassment tactics by creating lists that imply something horrible about everyone on it, then sharing the list without context. I've seen several of this type of list.

**Comment by legowerewolf** [comment]
> And where's the ability to remove oneself from lists? Plenty of users have pointed out that moderation lists can themselves be used as mass-harassment tactics by creating lists that imply something horrible about everyone on it, then sharing the list without context. I've seen several of this type of list.

This is a permission thing. The way Bluesky/atproto is structured, a moderation list is a record in a user's personal data repository, which only _they_ have write access to. There's no way for one user to update a record in another user's repository, and really, there shouldn't be.

Either way, it's off-topic. Make a new issue about it if you think you have an idea for how this could reasonably work.

**Comment by SnoillerFitzroy** [comment]
> This is a permission thing. The way Bluesky/atproto is structured, a moderation list is a record in a user's personal data repository, which only _they_ have write access to. There's no way for one user to update a record in another user's repository, and really, there shouldn't be.

Then it sounds like "Moderation" lists have been programmed in the single worst possible way. Maybe they should be programmed in a way that lets users avoid harassment from BOTH ends instead of this fascist dictatorship where you can't even control which reposts land in your feed and can't even have a private account and can't even EDIT a post?!?

**Comment by ohshazbot** [comment]
> Either way, it's off-topic. Make a new issue about it if you think you have an idea for how this could reasonably work.

we're here about the ability to hide resposts and quote posts per followed account, please respect that

**Comment by sax6575** [comment]
uh, why isn't this implemented yet?

**Comment by surfdude29** [comment]
The latest on this from one of the Bluesky devs:

> Muting reposts and/or quoteposts is close, hope to land soon 🙏

https://bsky.app/profile/esb.lol/post/3mr5zmpva6s2z

**Comment by mschwendener** [comment]
Just thought about this yesterday, thank you for the update!

**Comment by surfdude29** [comment]
Just merged 🥳🎉

- #11223

Coincidentally, three years to the day since this issue was opened 😅

(I think hiding quote posts per account is planned for a follow-up)

**Comment by aknightzero** [comment]
So does this mean this is live yet?

**Comment by surfdude29** [comment]
> So does this mean this is live yet?

It's live on the dev server: https://main.bsky.dev/

And it will be in the next version (1.130) of the iOS and Android apps and on bsky.app when that's released – probably in the next week or two, but the timing is of course up to the devs :)

**Comment by Mementh** [comment]
> > So does this mean this is live yet?
> 
> It's live on the dev server: https://main.bsky.dev/
> 
> And it will be in the next version (1.130) of the iOS and Android apps and on bsky.app when that's released – probably in the next week or two, but the timing is of course up to the devs :)

Will this let me hide or mute myself so i dont see my quotes and messges?

**Comment by mozzius** [comment]
Solved by #11223

**Comment by djb-rh** [comment]
> Additional request as part of this: there are accounts that I don't want to block all reposts, I just want to block when they repost THEMSELVES. So many in the media, in particular "repost for the morning crowd" something they already posted. I keep up with my entire timeline generally so I NEVER want to see a post twice. 

Sadly, this part still isn't implemented.
